### PR TITLE
fixed #1571

### LINKED ISSUE #1571: syntax for intersections
this should completely parallel unions (`and`). The concrete syntax should be `OMS intersect OMS`. 
See also #1570.

### PR DESCRIPTION
**Pull request by tillmo**

### CODE REVIEW

File: Syntax/AS_Structured.der.hs [hunk 1]
```diff
@@ -48,6 +48,8 @@ data SPEC = Basic_spec G_basic_spec Range
           | Bridge (Annoted SPEC) [RENAMING] (Annoted SPEC) Range
           | Union [Annoted SPEC] Range
             -- pos: "and"s
+          | Intersection [Annoted SPEC] Range
+            -- pos: "and"s
```
Comment (cmaeder) [inline Syntax/AS_Structured.der.hs]: "intersect"

**Comment by cmaeder** [comment]
the bracketing of multiple specs interspersed with "and" or "intersect" is somewhat unclear to me.
(maybe "intersect" should bind stronger?)

**Comment by cmaeder** [comment]
Ah, mixing "and" or "intersect" is forbidden. :+1:

File: Syntax/AS_Structured.der.hs (hunk 1 above)
Comment (tillmo) [inline Syntax/AS_Structured.der.hs]: OK, fixed

**Comment by tillmo** [comment]
yes, unless explicitly bracketed.

**Comment by cmaeder** [comment]
Hang on, I'm not sure if the pretty printer preserves braces of nested `and` and `intersect` blocks.

**Comment by tillmo** [comment]
probably not... @gnn, can you please take care of this?

**Comment by tillmo** [comment]
I have merged this into master (even though printing is not complete yet) because we need this feature.

**Pull request merged**